app/controllers/units_controller.rb

```diff
@@ -0,0 +1,35 @@
+class UnitsController < ApplicationController
+  def create
+    @system = System.find(params[:unit][:system_id])
```

Comment:
There shouldn't be anywhere to get there without those being defined, though. Is it a problem if it breaks for someone messing around nefariously?